doc/design/software-bridge-management.md

```diff
@@ -0,0 +1,460 @@
+* support configuration of software bridge in both modes (operator's `configurationMode` setting): `daemon` and `systemd`
+* implementation should be compatible with [_Externally Managed PF_](externally-manage-pf.md) feature
+
+
+### Non-Goals
+
+* replace `SriovNetworkPoolConfig` CRD
+* change API for host-level settings, e.g. `ovs-hw-offload`
+	
+	_**Note:** we may need to extend this API to support additional options_
+
+* add support for VF-lag use-case
+
+## Proposal
+
+1. deploy `ovs-cni` and `accelerated-bridge-cni` with init containers of `sriov-network-config-daemon` Pod
+2. define `OVSNetwork` and `BridgeNetwork` CRDs and implement controllers for them which will create `NetworkAttachmentDefinition` CRs
+3. extend `SriovNetworkNodePolicy`and `SriovNetworkNodeState` CRDs to support configuration of software bridges (bridge-level configuration)
+
+
+### Workflow Description
+
+Implementation should be compatible with the following workflows:
+
+* [Fully automatic workflow](#fully-automatic-workflow)
+* [NIC configuration only flow](#nic-configuration-only-flow) (Externally managed bridge)
+* [Externally managed NIC flow](#externally-managed-nic-flow)
+
+#### Fully automatic workflow
+
+This workflow assumes that **sriov-network-operator** handles PFs configuration, creation and configuration of software bridge, announcement of SRIOV resources with device plugin and preparation of `NetworkAttachmentDefinition` CR.
+
+1. User creates `SriovNetworkNodePolicy` where:
+    * eswitch mode set to `switchdev`
+    * configuration for selected software bridge is defined
+
+2. The operator populates `SriovNetworkNodeState` for matching nodes with PF and bridge
+configurations 
+
+3. `sriov-network-config-daemon` applies PF configuration, create and configure bridge, attach PF to the bridge, applies VF configuration
+
+
+    _**Note 1:** if the operator runs in the `systemd` mode then bridge creation should happen in the `pre` phase._
+
+    _**Note 2:** we should create udev rule which will set `NM_UNMANAGED=1` for bridges created by the operator_
+
+4. `sriov-network-config-daemon` should report information about software bridges in the status field of the `SriovNetworkNodePolicy` CR.
+
+5. SRIOV resources are announced by the Device plugin
+
+6. User creates `OVSNetwork` or `BridgeNetwork` CR to create `NetworkAttachmentDefinition` CR, which relies on resources announced by the Device Plugin
+
+
+_**Note:** created software bridge should be removed during the PF configuration reset_
+
+
+#### NIC configuration only flow
+
+This workflow is kept to support existing HW offloading use-case. 
+
+In this case, **sriov-network-operator** handles PFs configuration, announcement of SRIOV resources with device plugin and preparation of `NetworkAttachmentDefinition` CR.
+
+In some scenarios, it may be compatible with `configurationMode: daemon`, but it is supposed to be used when the operator runs in `systemd` mode.
```

Comment:
> when would it not be compatible with daemon mode ?

When `NIC configuration only flow` is used with `systemd` configurationMode, then you can apply any network configuration between  steps when VF is created  and when VF is configured(VF binds to the driver). In the `daemon` mode you don't have such option, you can only apply your custom network configuration before the operator or after, but not in the middle. This can be a problem for some configuration, e.g. for VF LAG for NVIDIA NICs